core/src/main/java/org/lfenergy/shapeshifter/core/service/validation/UftpMessageSupport.java

```diff
@@ -25,38 +25,42 @@
 public interface UftpMessageSupport {
 
   /**
-   * Gets a previously received message by messageID (if any).
+   * Gets a previously received message by messageID (if any) and conversation ID.
    *
    * @param messageID The message ID.
+   * @param conversationId The conversation ID
    * @param recipientDomain The recipient domain.
    * @return The message that was received previously.
    */
-  Optional<PayloadMessageType> getPreviousMessage(String messageID, String recipientDomain);
+  Optional<PayloadMessageType> getPreviousMessage(String messageID, String conversationId, String recipientDomain);
```

Comment:
After further thinking: this method is used to check for duplicate MessageIds which must be unique within a sender not within a conversation, so I think actually conversationID should not be there but senderDomain should be added.

Reply:
changed to `findDuplicateMessage`